packages/smooth_app/lib/pages/offline_data.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_list_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class OfflineDataScreen extends StatefulWidget {
+  const OfflineDataScreen();
+  @override
+  State<OfflineDataScreen> createState() => _OfflineDataScreenState();
+}
+
+int length = 0;
+double size = 0;
+
+// TODO(ashaman999): update all the applocalizations
+class _OfflineDataScreenState extends State<OfflineDataScreen> {
+  Future<void> _getDetails(DaoProduct daoProduct) async {
+    length = await daoProduct.getLength() ?? 0;
+    size = await daoProduct.getSize();
+    setState(() {});
```

Comment:
I would prefer working with a FutureBuilder here

Reply:
did the changes